src/main/java/bio/terra/workspace/service/datarepo/DataRepoService.java

```diff
@@ -1,47 +1,32 @@
 package bio.terra.workspace.service.datarepo;
 
+import bio.terra.datarepo.api.RepositoryApi;
+import bio.terra.datarepo.client.ApiClient;
+import bio.terra.datarepo.client.ApiException;
 import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
-import org.springframework.boot.web.client.RestTemplateBuilder;
-import org.springframework.context.annotation.Bean;
-import org.springframework.http.HttpEntity;
-import org.springframework.http.HttpHeaders;
-import org.springframework.http.HttpMethod;
-import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Component;
-import org.springframework.web.client.RestClientException;
-import org.springframework.web.client.RestTemplate;
 
 @Component
 public class DataRepoService {
 
-  private RestTemplate restTemplate;
-
-  @Bean
-  private RestTemplate restTemplate(RestTemplateBuilder builder) {
-    return builder.build();
-  }
-
-  public DataRepoService() {
-    this.restTemplate = new RestTemplate();
+  private ApiClient getApiClient(String accessToken) {
+    ApiClient client = new ApiClient();
+    client.setAccessToken(accessToken);
+    return client;
   }
 
-  private String getSnapshotUrl(String instance, String snapshotId) {
-    return instance + "/api/repository/v1/snapshots/" + snapshotId;
+  private RepositoryApi repositoryApi(String instance, AuthenticatedUserRequest userReq) {
+    return new RepositoryApi(getApiClient(userReq.getRequiredToken()).setBasePath(instance));
   }
 
   public boolean snapshotExists(
       String instance, String snapshotId, AuthenticatedUserRequest userReq) {
-    try {
-      HttpHeaders headers = new HttpHeaders();
-      headers.setBearerAuth(userReq.getRequiredToken());
-
-      HttpEntity<String> entity = new HttpEntity<>(headers);
-      ResponseEntity response =
-          restTemplate.exchange(
-              getSnapshotUrl(instance, snapshotId), HttpMethod.GET, entity, Object.class);
+    RepositoryApi repositoryApi = repositoryApi(instance, userReq);
 
-      return response.getStatusCode().is2xxSuccessful();
-    } catch (RestClientException e) {
+    try {
+      repositoryApi.retrieveSnapshot(snapshotId);
```

Comment:
this throws an `ApiException` if the snapshot is not found/user does not have permission, correct?  I believe that's true but want to confirm.

Reply:
yes it should throw for all non-successful status codes, which would cover this case